rendercanvas/_loop.py

```diff
@@ -58,6 +59,7 @@ def __init__(self):
         self.__state = (
             0  # 0: off, 1: ready, 2: detected-active, 3: inter-active, 4: running
         )
+        self._asyncgens = weakref.WeakSet()
```

Comment:
If possible to avoid weak refs of any kind, please do :) especially in hot code paths like event loops, do we really want to have them?

Reply:
My plan is to replicate what asyncio does, and it does use a weakref ;)  but note that this code only runs before a loop start, not on every step.